channeldb/channel.go

```diff
@@ -248,6 +248,10 @@ type chanAuxData struct {
 	// memo is an optional text field that gives context to the user about
 	// the channel.
 	memo tlv.OptionalRecordT[tlv.TlvType5, []byte]
+
+	// tapscriptRoot is the optional Tapscript root the channel funding
+	// output commits to.
+	tapscriptRoot tlv.OptionalRecordT[tlv.TlvType6, [32]byte]
```

Comment:
The extra data is only used for serialization and de-serialization. It houses all the fields that are new and stored in a TLV stream (as opposed to the old `lnwire.WriteElements()` way). If we have the fields in a struct, we can use the new `tlv.RecordT` and `tlv.OptionalRecordT` types, which is nice. But because we don't want to refactor everything to access those records, we still copy from/to the `OpenChannel` struct. I renamed this to `openChannelTlvData` and added a comment to make this a bit more clear.